geographer/downloaders/eodag_downloader_for_single_vector.py

```diff
@@ -0,0 +1,413 @@
+            # sorting by multiple keys.
+            key, asc_or_desc = sort_by[0]
+            reverse = {"ASC": False, "DESC": True}.get(asc_or_desc)
+            result = SearchResult(
+                products=sorted(
+                    result, key=lambda product: product.properties[key], reverse=reverse
+                ),
+                number_matched=result.number_matched,
+                errors=result.errors,
+            )
+
+        # Return dicts with values to be collected in calling associator.
+        raster_info_dict = {}
+
+        for eo_product in result:
+
+            # For the next couple of lines we are essentially following
+            # the _prepare_download method of the
+            # eodag.plugins.download.base.Download class to extract
+            # the name of the extracted product.
+            sanitized_title = sanitize(eo_product.properties["title"])
+            if sanitized_title == eo_product.properties["title"]:
+                collision_avoidance_suffix = ""
+            else:
+                collision_avoidance_suffix = "-" + sanitize(eo_product.properties["id"])
+            extracted_product_file_name = sanitized_title + collision_avoidance_suffix
+
+            if suffix_to_remove is not None:
+                raster_name = (
+                    extracted_product_file_name.removesuffix(suffix_to_remove) + ".tif"
+                )
+            else:
+                raster_name = extracted_product_file_name + ".tif"
+
+            if raster_name not in previously_downloaded_rasters_set:
+
+                download_params = download_kwargs | dict(
+                    product=eo_product,
+                    output_dir=download_dir,
+                    extract=True,
+                )
+
+                try:
+                    location = self.eodag.download(**download_params)
+
+                    location_name = Path(location).name
+                    if location_name != extracted_product_file_name:
+                        msg = (
+                            "The name of the downloaded file (%s) does not "
+                            "match the expected name (%s). eodag must have "
+                            "changed the way they determine the file name. "
+                            "Unfortunately, `geographer` relies on being able "
+                            "to determine the name of the extracted file "
+                            "without downloading the product. The "
+                            "`EodagDownloaderForSingleVector` will have to be "
+                            "updated to work with the new naming convention of"
+                            "eodag. Sorry!"
+                        )
+                        log.error(msg, location_name, extracted_product_file_name)
+                        raise RuntimeError(
+                            msg % (msg, location_name, extracted_product_file_name)
+                        )
```

Comment:
I think it would make sense to define at least one base error class (GeographerError) to use here.

Also, it's not clear to me why you log an error _and_ raise an exception. If you don't catch the exception, then the error message is displayed anyway. The point of logging would be to display the message anyway, even if the exception is caught. Might or might not make sense in this case, I don't know.

Reply:
I think I was confused about how to do error handling, and honestly still am...

What is the general principle for which exceptions are inherited from `GeoGrapherError`?